portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -435,25 +409,77 @@ private void lock(SecurityAdvisor securityAdvisor) {
     }
 
     private void doAcademicInsert(String from, String to, String event, String ref
-                                            , String title, String siteId, Date eventTime ,String url) {
+                                            , String title, String siteId, Date eventDate ,String url) {
 
-        sqlService.dbInsert(null
-            , BULLHORN_INSERT_SQL
-            , new Object[] {ACADEMIC, from, to, event, ref, title, siteId, eventTime, url}
-            , "ID");
+        Session session;
+        Transaction tx = null;
+        try {
+            session = sessionFactory.getCurrentSession();
```

Comment:
No need for this to be in a try block, also you could use the EntityManager instead of the session here.